src/Symfony/Component/HttpKernel/Controller/ArgumentValueResolver/ArgumentIsRequest.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Controller\ArgumentValueResolver;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Controller\ArgumentValueResolverInterface;
+use Symfony\Component\HttpKernel\ControllerMetadata\Argument\ArgumentMetadataInterface;
+
+/**
+ * Supports the same instance as the request object passed along.
+ *
+ * @author Iltar van der Berg <kjarli@gmail.com>
+ */
+final class ArgumentIsRequest implements ArgumentValueResolverInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function supports(Request $request, ArgumentMetadataInterface $argument)
+    {
+        return $argument->getArgumentType() === get_class($request);
```

Comment:
The current check is doing an instanceof check via reflection. I agree that your solution might provide a more secure check. Changing it to:
```php
return $argument->getArgumentType() === Request::class || is_subclass_of($request, $argument->getArgumentType());
```